src/mono/mono/metadata/appdomain.c

```diff
@@ -1255,8 +1255,8 @@ mono_runtime_register_appctx_properties (int nprops, const char **keys,  const c
 	for (int i = 0; i < nprops; ++i) {
 		appctx_keys [i] = g_new0 (char, strlen (keys [i]) + 1);
```

Comment:
You still have these allocations (which are immediately overwritten).

Reply:
Fixed. Thanks!